File: weeak-1/maximumPairWiseProduct-notBestSolution/maximumPairWiseProduct.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter Array of integers ", function(arr){
    arr = arr.split(" ")
    arr = arr.map(x => parseInt(x))
    console.log(sumProduct(arr))
})

rl.on("close", function() {
    console.log("\nBYE BYE !!!");
    process.exit(0);
});

function sumProduct(arr){
 let sum = 0;
 for(let i=0; i<arr.length; i++){
	 for(let j=i+1; j<arr.length; j++){
		 if(arr[i] * arr[j] > sum){
			 sum = arr[i] * arr[j]
		 }
	 }
    }
  return sum
}

File: weeak-1/maximumPairWiseProduct-o(n^2)/maximumPairWiseProduct.py
arr = input("Enter Array ").split(" ")
arr = list(map(int, arr))

def getMaxProduct(arr):
    sum = 0
    count = 0;
    for value1 in arr:
        count += 1
        for value2 in arr[count::]:
            if value1 * value2 > sum:
                sum = value1 * value2 
    return sum

print(getMaxProduct(arr));
